packages/graphback/src/resolvers/index.ts

```diff
@@ -19,20 +20,20 @@ export class ResolverGenerator {
     this.inputContext = inputContext
   }
 
-  public generate(database: string) {
+  public generate(database: string, templateType: string) {
     this.context = buildResolverTargetContext(this.inputContext, database)
 
-    const customContext = createCustomContext(this.inputContext)
+    const customContext = createCustomContext(this.inputContext, templateType)
 
     let hasCustomElements = false
     if(customContext.length) {
       hasCustomElements = true
     }
 
-    return {
-      types: generateResolvers(this.context),
-      index: generateIndexFile(this.context, hasCustomElements),
-      custom: generateCustomResolvers(customContext)
+    if(templateType === 'graphql-js') {
+      return generateGraphQLjsResolvers(this.context, customContext, hasCustomElements)
+    } else {
+      return generateApolloResolvers(this.context, customContext, hasCustomElements)
```

Comment:
According to the template

Reply:
Yes. This is quite tricky to test and makes the generator much more complex. 
Let's imagine that we want to generate a different format than knex and in this generator, we will need to also have template type covered. This change makes things much more complex and less contributor friendly. Our success depends on extensibility - if it is hard to extend